swiftpkg/internal/swiftpkg_build_files.bzl

```diff
@@ -51,6 +51,16 @@ def _swift_target_build_file(pkg_ctx, target):
         "srcs": pkginfo_targets.srcs(target),
         "visibility": ["//visibility:public"],
     }
+
+    # Add macros as plugins
+    macro_target_labels = [target.label.repository_name + "//:" + target.label.name for target in pkg_ctx.pkg_info.targets if target.type == "macro"]
+    if macro_target_labels:
+        plugins = [target_label for target_label in macro_target_labels for dep in deps if target_label in dep.value[0]]
+        if plugins:
+            attrs["plugins"] = plugins
+            deps_without_plugins = [dep for dep in deps if dep.value[0] not in plugins]
+            attrs["deps"] = bzl_selects.to_starlark(deps_without_plugins)
```

Comment:
Suggestions for how to refactor this are appreciated.

Reply:
It looks reasonable to me. Unfortunately, the API for assembling Bazel select statements is a little convoluted due to how they need to be gathered and merged.